packages/openapi3/src/schema-emitter.ts

```diff
@@ -653,6 +660,180 @@ export class OpenAPI3SchemaEmitter extends TypeEmitter<
     }
   }
 
+  #attachXmlObjectForScalarOrModel(
+    program: Program,
+    prop: Scalar | Model,
+    emitObject: OpenAPI3Schema,
+  ) {
+    const xmlObject: OpenAPI3XmlSchema = {};
+
+    // Resolve XML name
+    const xmlName = resolveEncodedName(program, prop, "application/xml");
+    if (xmlName !== prop.name) {
+      xmlObject.name = xmlName;
+    }
+
+    // Get and set XML namespace if present
+    const currNs = getNs(program, prop);
+    if (currNs) {
+      xmlObject.prefix = currNs.prefix;
+      xmlObject.namespace = currNs.namespace;
+    }
+
+    // Attach xml schema to emitObject if not empty
+    if (Object.keys(xmlObject).length !== 0) {
+      emitObject.xml = xmlObject;
+    }
+  }
+
+  #attachXmlObjectForModelProperty(
+    program: Program,
+    prop: ModelProperty,
+    emitObject: OpenAPI3Schema,
+    ref?: Record<string, any>,
+  ) {
+    const xmlObject: OpenAPI3XmlSchema = {};
+
+    const isXmlModel = this.#isXmlModelChecker(program, prop.model!, []);
+    if (!isXmlModel) {
+      return;
+    }
+
+    // Resolve XML name
+    const xmlName = resolveEncodedName(program, prop, "application/xml");
+    const jsonName = resolveEncodedName(program, prop, "application/json");
+    if (xmlName !== prop.name && xmlName !== jsonName) {
+      xmlObject.name = xmlName;
+    }
+
+    // Get and set XML namespace if present
+    const currNs = getNs(program, prop);
+    if (currNs) {
+      xmlObject.prefix = currNs.prefix;
+      xmlObject.namespace = currNs.namespace;
+    }
+
+    // Set XML attribute if present
+    if (isAttribute(program, prop)) {
+      if (prop.type?.kind === "Model") {
+        reportDiagnostic(program, {
+          code: "invalid-property-type",
+          format: { name: prop.name },
+          target: prop,
+        });
+      } else {
+        xmlObject.attribute = true;
+      }
+    }
+
+    // Handle array wrapping if necessary
+    const isArrayProperty = prop.type?.kind === "Model" && isArrayModelType(program, prop.type);
+    const hasUnwrappedDecorator = isUnwrapped(program, prop);
+    if (!isArrayProperty && hasUnwrappedDecorator) {
+      reportDiagnostic(program, {
```

Comment:
this is not true, unwrap applies to primitive types as well. It means to add that to the raw node text

Reply:
in those case it should have a better dedicated message explaining the feature gap in openapi3